components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -63,11 +63,12 @@ function getUserCacheKey(id: string): string {
 @injectable()
 export class TypeORMUserDBImpl extends TransactionalDBImpl<UserDB> implements UserDB {
     constructor(
-        @inject(EncryptionService) protected readonly encryptionService: EncryptionService,
-        @inject(DataCache) protected readonly cache: DataCache,
-        @optional() transactionalEM: EntityManager,
+        @inject(TypeORM) typeorm: TypeORM,
```

Comment:
Where do you see the benefit in having it listed here as explicit parameter, beyond the aspirational "it would be nice to only use ctor injection"? What's the concrete use-case? :thinking:

Reply:
Not being surprised when 
- using the constructor, e.g. in unit tests. 
- using the typeorm field in an otherwise constructed object (i.e. transactional or unit test)

Reply:
Ah, now I understand your comment from two days ago :smile: . I think I never ran into a case where I had to construct these so I was wondering.